Release/1.7.0

mike-kelly-methods opened the pull request:
### What

`dp-identity-api` release `1.7.0` which includes adding a new `aws cognito` user to a user pool functionality.

### How to review

Ensure all unit and feature tests pass.

### Who can review

Anyone but me and @sandypadmanabhan.

daiLlew commented on config/config.go:
```diff
@@ -19,9 +19,11 @@ type Configuration struct {
+	BindAddr           string `envconfig:"MONGODB_BIND_ADDR"   json:"-"`
+	Collection         string `envconfig:"MONGODB_COLLECTION"`
+	IdentityCollection string `envconfig:"MONGODB_IDENTITY_COLLECTION"`
+	TokenCollection    string `envconfig:"MONGODB_TOKEN_COLLECTION"`
```
TODO - will need to update app config secrets with new fields.

daiLlew commented on mongo/token_store.go:
```diff
@@ -3,13 +3,106 @@ package mongo
+	return active, nil
+}
+
+func (m *Mongo) deleteActiveTokens(ctx context.Context, i schema.Identity, active []schema.Token) error {
```
Might be better to just call this `deleteTokens` it doesn't need to know that they are specifically "active tokens".

eldeal commented on mongo/token_store.go:
```diff
@@ -3,13 +3,101 @@ package mongo
+// will be marked as deleted. Sets the last modified date on all documents updated.
 func (m *Mongo) StoreToken(ctx context.Context, tkn schema.Token, i schema.Identity, ttl time.Duration) error {
+	logD := log.Data{identityIDKey: i.ID}
+	log.InfoCtx(ctx, "tokenStore: storing identity token", logD)
```
I'm not sure I'd put `tokenStore` in all of these log messages. The messages should be descriptive enough on their own, and context should allow us to aggregate. But I'm aware its relatively subjective.

sandypadmanabhan committed:
WIP for adding post user endpoint

sandypadmanabhan committed:
WIP for adding post user endpoint

sandypadmanabhan committed:
resolved conflicts

sandypadmanabhan committed:
resolved conflicts

sandypadmanabhan committed:
updated code to generate password

sandypadmanabhan committed:
Added test for user.go

mike-kelly-methods committed:
Finish unit test and add key to CreateUserOutput model struct.

mike-kelly-methods committed:
Finish unit test and add key to CreateUserOutput model struct.

mike-kelly-methods committed:
Move error handling routines to apierrors.go package for service wide usage.

sandypadmanabhan committed:
Added feature test to return 201 for user

sandypadmanabhan committed:
resolved conflicts

sandypadmanabhan committed:
Added response status 500 feature test to user

sandypadmanabhan committed:
Added user endpoint to swagger spec

mike-kelly-methods committed:
Adding username and email validation checks and unit test.

mike-kelly-methods committed:
Adding username and email validation checks and unit test.

sandypadmanabhan committed:
Merge branch 'develop' into feature/create_user_identity_api

sandypadmanabhan committed:
updated swagger for user endpoint for 400 error response

sandypadmanabhan committed:
resolved conflicts

mike-kelly-methods committed:
Add logging to errors, add constants to error outputs and new feature tests.

mike-kelly-methods committed:
Merge branch 'develop' into feature/create_user_identity_api

mike-kelly-methods committed:
Updated error handling in users.go, unit and feature tests.

mike-kelly-methods committed:
Fix feature tests by setting dummy pool id.

mike-kelly-methods committed:
Add duplicate email checkt to create user handler.

mike-kelly-methods committed:
Add duplicate email check, unit and fetaure tests.

mike-kelly-methods committed:
Merge branch 'develop' into feature/create_user_identity_api

mike-kelly-methods committed:
Fix tokens feature test - casing.

mike-kelly-methods committed:
PR comments - add handling for forename and surname in request body and tests.

mike-kelly-methods committed:
Merge develop to fix conflict.

mike-kelly-methods committed:
Add check for required params at api setup.

mike-kelly-methods committed:
Merging develop - fixing conflicts.

mike-kelly-methods committed:
PR comments - code tity up and fix swagger.yaml.

mike-kelly-methods committed:
Merge branch 'feature/create_user_identity_api' into develop

mwnicks committed:
add missing check to api_test

mwnicks committed:
add unit tests for the refresh tokens endpoint

mwnicks committed:
add feature test scenarios for the refresh token endpoint

mwnicks committed:
add cognito stub/mock for tests

mwnicks committed:
add id token mocker and feature step

mwnicks committed:
fix feature test definitions

mwnicks committed:
add error handling for missing tokens

mwnicks committed:
add ID token parser

mwnicks committed:
add error handling logic for unparsable ID tokens

mwnicks committed:
move log inside handler function so it doesn't log on start up

mwnicks committed:
fix typo

mwnicks committed:
implement error handling for Cognito error responses

HeJones committed:
Merge branch 'develop' into feature/update-login-endpoint

mwnicks committed:
fix failing tests

mwnicks committed:
add unit tests for the ID model

mwnicks committed:
refactor validation in to models

mwnicks committed:
update swagger docs

janderson2 committed:
Move current api error handling aside

Rename the current api error handling package to make room for the
refinement work so that we can iteratively update the endpoints rather
than being forced to do them in one big sweep.

janderson2 committed:
Remove the hello world example endpoint

janderson2 committed:
Format the codebase to align with standard style

Run `go fmt` on the code base.

janderson2 committed:
Update error response model inline with new standard

Update the error response model to:

* Remove the `source` block
* Rename `error` field to `code`
* Rename `message` field to `description`

mwnicks committed:
fix merge conflicts

mwnicks committed:
address PR comments

HeJones committed:
Add admin logout call before login

HeJones committed:
Merge branch 'develop' into feature/update-login-endpoint

HeJones committed:
Add in new error package

mwnicks committed:
Merge branch 'feature/refresh-tokens' into develop

HeJones committed:
update function name for clarity

HeJones committed:
Merge branch 'develop' into feature/update-login-endpoint

HeJones committed:
update error package

HeJones committed:
Merge branch 'feature/update-login-endpoint' into develop

HeJones reviewed (approved):
It looks good to me
